sql/src/main/java/io/crate/action/sql/SQLResponse.java

```diff
@@ -60,25 +66,35 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         builder.startObject();
         writeSharedAttributes(builder);
         builder.startArray(Fields.ROWS);
-        if (rows != null) {
-            for (Object[] row : rows) {
-                builder.startArray();
-                for (int j = 0, len = cols().length; j < len; j++) {
-                    builder.value(row[j]);
-                }
-                builder.endArray();
+        for (Row row : bucket) {
+            builder.startArray();
+            for (int j = 0, len = cols().length; j < len; j++) {
```

Comment:
That doesn't work, This is iteratoring over the columns of the Row - not over the column names.
And ``Row`` doesn't extend ``Iterable<Object>``